pipelines/controller/src/main/java/org/openmrs/analytics/metrics/PipelineMetricsEndpoint.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020-2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openmrs.analytics.metrics;
+
+import io.prometheus.client.Collector.MetricFamilySamples;
+import io.prometheus.client.CollectorRegistry;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Enumeration;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.openmrs.analytics.MetricsConstants;
+import org.springframework.boot.actuate.endpoint.Producible;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.web.WebEndpointResponse;
+import org.springframework.boot.actuate.endpoint.web.annotation.WebEndpoint;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+import org.springframework.util.MimeType;
+import org.springframework.util.MimeTypeUtils;
+
+/**
+ * This is a custom implementation of the spring actuator {@link Endpoint @Endpoint} which exports
+ * only the pipeline metrics in a format that can be scraped by the Prometheus server.
+ */
+@WebEndpoint(id = "pipeline-metrics")
+@Component
+public class PipelineMetricsEndpoint {
```

Comment:
The end point exposed `/actuator/pipeline-metrics` will only be used in the pipeline controller UI because the output format is little simplified for human readability. However, the regular endpoints like `/actuator/prometheus`, `/actuator/metrics` remains unmodified and can be used by different monitoring systems.

Reply:
I see thanks for the clarification and also for the simplifications.